app/services/daily_digest_service.rb

```diff
@@ -10,13 +10,10 @@ def execute
     updates = questions_from_today
     updates = add_questions_with_no_activity(updates)
 
-    User.joins(:communities).includes(:communities, :school).joins(:startups)
+    User.where.not(communities: nil).or(User.where.not(faculty: nil))
```

Comment:
This does not work.

```suggestion
    students = User.joins(:communities).distinct.select(:id)
    coaches = User.joins(:faculty).select(:id)
    User.where(id: coaches).or(User.where(id: students))
```

Also use `find_each` instead of `each` to batch the loading of `User` objects.

@mahesh-krishnakumar Note!